src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -795,14 +835,402 @@ def __try_add_error(error_list, detail):
+
+                if len(packages_removed_from_installation) > 0:
+                    # Update current installation # of removed packages
+                    self.__installation_packages_removed = packages_removed_from_installation
+                    # Recompose truncated status file payload (installation)
+                    truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file, packages_truncated_in_installation, self.__installation_total_error_count,
+                        self.__installation_substatus_msg_copy, Constants.PATCH_INSTALLATION_SUMMARY, installation_substatus_index)
+
+                status_file_size_in_bytes, current_package_list_size, packages_capacity_diff = self.__get_new_size_in_bytes_after_truncation(truncated_status_file, max_package_list_capacity, packages_truncated_in_assessment, packages_truncated_in_installation)
+                current_package_list_size -= packages_capacity_diff
+
+            self.composite_logger.log_debug("End package list truncation")
+
+        # Write status file
+        self.env_layer.file_system.write_with_retry_using_temp_file(self.status_file_path, '[{0}]'.format(json.dumps(truncated_status_file)), mode='w+')
+
+    def __get_new_size_in_bytes_after_truncation(self, truncated_status_file, max_package_list_capacity, assessment_package_list, installation_package_list):
+        """ Get new size in bytes for status_file, package_list, and diff after 1xxx truncation """
+        current_package_list_size = self.__get_current_packages_size_in_bytes(assessment_package_list, installation_package_list)
+
+        status_file_size = self.__calc_status_size_on_disk(truncated_status_file)
+        truncated_package_list_diff = current_package_list_size - max_package_list_capacity
+
+        return status_file_size, current_package_list_size, truncated_package_list_diff
+
+    def __get_current_packages_size_in_bytes(self, assessment_package_list, installation_package_list):
+        """ Get current complete assessment, installation, or both package list size in bytes """
+        # Only installation data exists, get installation list capacity
+        if not assessment_package_list:
+            return self.__calc_package_payload_size_on_disk(installation_package_list)
+
+        # Only assessment data exists, get assessment list capacity
+        if not installation_package_list:
+            return self.__calc_package_payload_size_on_disk(assessment_package_list)
+
+        # Both assessment and installation data exist
+        return self.__calc_package_payload_size_on_disk(assessment_package_list) + self.__calc_package_payload_size_on_disk(installation_package_list)
+
+    def __cut_list_and_get_list_capacity(self, package_list, max_package_list_capacity):
+        """ Split package list, keep 5 minimum packages, and remaining packages for truncation """
```

Comment:
Why are we calculating sizes here? IIUC, the main aim of this function is split assessment list into 2 parts, where the 1 part contains 5 packages and the 2nd one contains remaining.

This function is not performing any truncation related activity, simply dividing a list into 2. Let's keep it to that and keep all size computations restricted to the truncation function

Reply:
logic move to __apply_truncation_process()